core/drivers/crypto/se050/adaptors/apis/apdu.c

```diff
@@ -0,0 +1,685 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <fsl_sss_api.h>
+#include <scp.h>
+#include <se050_apdu_apis.h>
+#include <string.h>
+
+sss_status_t se050_factory_reset(pSe05xSession_t ctx)
+{
+	smStatus_t st = SM_OK;
+
+	if (!ctx)
+		return kStatus_SSS_Fail;
+
+	st = Se05x_API_DeleteAll_Iterative(ctx);
+	if (st != SM_OK)
+		return kStatus_SSS_Fail;
+
+	return kStatus_SSS_Success;
+}
+
+sss_status_t se050_cipher_update_nocache(sss_se05x_symmetric_t *ctx,
+					 const uint8_t *src, size_t src_len,
+					 uint8_t *dst, size_t *dst_len)
+{
+	smStatus_t status = SM_NOT_OK;
+
+	if (!ctx || !src || !dst || !dst_len)
+		return kStatus_SSS_Fail;
+
+	status = Se05x_API_CipherUpdate(&ctx->session->s_ctx,
+					ctx->cryptoObjectId,
+					src, src_len,
+					dst, dst_len);
+	if (status != SM_OK)
+		return kStatus_SSS_Fail;
+
+	return kStatus_SSS_Success;
+}
+
+uint8_t se050_key_exists(uint32_t key_id, pSe05xSession_t ctx)
+{
+	SE05x_Result_t inuse = kSE05x_Result_FAILURE;
+	smStatus_t status = SM_OK;
+
+	if (!ctx)
+		return 0;
+
+	status = Se05x_API_CheckObjectExists(ctx, key_id, &inuse);
+	if (status != SM_OK)
+		return 0;
+
+	if (inuse == kSE05x_Result_SUCCESS)
+		return 1;
+
+	return 0;
+}
+
+sss_status_t se050_key_store_set_rsa_key_bin(sss_se05x_key_store_t *store,
+					     sss_se05x_object_t *k_object,
+					     struct rsa_keypair_bin *key_pair,
+					     struct rsa_public_key_bin *key_pub,
+					     size_t key_bit_len)
+{
+	SE05x_RSAKeyFormat_t rsa_format = kSE05x_RSAKeyFormat_RAW;
+	SE05x_TransientType_t type = kSE05x_TransientType_Transient;
+	Se05xSession_t *s_ctx = NULL;
+	uint32_t key_type = 0;
+	Se05xPolicy_t policy = {
+		.value = NULL, .value_len = 0,
+	};
+	sss_status_t ret = kStatus_SSS_Fail;
+	smStatus_t status = SM_NOT_OK;
+
+	if (!store || !store->session || !k_object)
+		return kStatus_SSS_Fail;
+
+	s_ctx = &store->session->s_ctx;
+	key_type = k_object->objectType;
+	if (k_object->isPersistant)
+		type = kSE05x_TransientType_Persistent;
+
+	switch (k_object->cipherType) {
+	case kSSS_CipherType_RSA:
+		rsa_format = kSE05x_RSAKeyFormat_RAW;
+		break;
+	case kSSS_CipherType_RSA_CRT:
+		rsa_format = kSE05x_RSAKeyFormat_CRT;
+		break;
+	default:
+		ret = kStatus_SSS_Fail;
+		goto exit;
+	}
+
+	if (se050_key_exists(k_object->keyId, s_ctx))
+		key_bit_len = 0;
+
+	if (key_type != kSSS_KeyPart_Public)
+		goto label_private;
```

Comment:
These `goto label_*` hides a switch/case case. I think it would be easier to maintain using a switch case.
Since the `SM_OK` status is expected at each step, I would suggest something like:

```c
	switch (key_type) {
	case kSSS_KeyPart_Public:
		status = Se05x_API_WriteRSAKey(s_ctx, &policy, k_object->keyId,
						...);
		if (status != SM_OK)
			break;

		status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
						...);
		break;

	case kSSS_KeyPart_Private:
		switch (k_object->cipherType) {
		case kSSS_CipherType_RSA:
			status = Se05x_API_WriteRSAKey(s_ctx, &policy, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			break;

		case kSSS_CipherType_RSA_CRT:
			status = Se05x_API_WriteRSAKey(s_ctx, &policy, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			break;

		default:
			return kStatus_SSS_Fail;
		}
		break;

	case kSSS_KeyPart_Pair:
		switch (k_object->cipherType) {
		case kSSS_CipherType_RSA;
			status = Se05x_API_WriteRSAKey(s_ctx, &policy, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			break;

		case kSSS_CipherType_RSA_CRT:
			status = Se05x_API_WriteRSAKey(s_ctx, &policy, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			if (status != SM_OK)
				break;

			status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
							...);
			break;

		default:
			return kStatus_SSS_Fail;
		}
		break;

	default:
		return kStatus_SSS_Fail;
	}

	if (status != SM_OK)
		return kStatus_SSS_Fail;

	return kStatus_SSS_Success;
}

Reply:
that was the first thing I did but throwing away the extra TAB  made the code unreadable when we have to respect the 80 character lines.